mountpoint-s3/examples/upload_benchmark.rs

```diff
@@ -107,11 +106,41 @@ fn main() {
     let runtime = client.event_loop_group();
 
     for i in 0..args.iterations {
+        let max_memory_target = if let Some(target) = args.max_memory_target {
+            target * 1024 * 1024
+        } else {
+            // Default to 95% of total system memory
+            let sys = System::new_with_specifics(RefreshKind::everything());
+            (sys.total_memory() as f64 * 0.95) as u64
+        };
+        let mem_limiter = Arc::new(MemoryLimiter::new(client.clone(), max_memory_target));
+
+        let buffer_size = args.write_part_size;
+        let server_side_encryption = ServerSideEncryption::new(args.sse.clone(), args.sse_kms_key_id.clone());
+
+        let checksum_algorithm = match args.checksum_algorithm.as_str() {
+            "off" => None,
+            "crc32c" => Some(ChecksumAlgorithm::Crc32c),
+            "crc32" => Some(ChecksumAlgorithm::Crc32),
+            "sha1" => Some(ChecksumAlgorithm::Sha1),
+            "sha256" => Some(ChecksumAlgorithm::Sha256),
+            other => Some(ChecksumAlgorithm::Unknown(other.to_string())),
+        };
```

Comment:
I would have preferred we just used an enum for `Args::checksum_algorithm`, `str` seems like a mistake.

Reply:
Yes, will fix separately, though.